src/OrleansTestingHost/TestCluster.cs

```diff
@@ -415,7 +421,8 @@ public void InitializeClient()
             GrainClient.Initialize(clientConfig);
             this.GrainFactory = GrainClient.GrainFactory;
             this.InternalGrainFactory = this.GrainFactory as IInternalGrainFactory;
-            this.StreamProviderManager = RuntimeClient.Current.CurrentStreamProviderManager;
+            this.ServiceProvider = GrainClient.ServiceProvider;
+            this.StreamProviderManager = this.ServiceProvider.GetRequiredService<IRuntimeClient>().CurrentStreamProviderManager;
```

Comment:
Isn't the StreamProviderManager stored directly in the service provider? As I mentioned before, I see IRuntimeClient as a glorified SP facade, with some extra features. But stuff like the stream provider manager looks like it can come directly from the SP.